scipy/sparse/sputils.py

```diff
@@ -302,6 +302,8 @@ def check_shape(args, current_shape=None):
             new_shape = new_shape[0:skip] + (unspecified,) + new_shape[skip+1:]
         else:
             raise ValueError('can only specify one unknown dimension')
+    elif len(new_shape) != 2:
```

Comment:
How does this interact with the checks starting on line 312?

Reply:
Well the current code will raise an IndexError before it gets to line 312 if this condition fails. This just makes the error explicit.

The logic at line 312 doesn't apply to np.matrix.resize, but the error there is weird.

Reply:
Yeah, it seems like line 312 is only reached when `current_shape` is provided, which isn't clear from the layout of the code. If you want to refactor this function I wouldn't mind, but we can also just push that off to another PR.

Reply:
It is reached, but doesn't do anything. I've reorganized the function. Sorry for being lazy earlier!

Reply:
Looks much better!